File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int sLength = s.length();
        int tLength = t.length();

        int l = 0;
        int r = 0;

        while(l < sLength && r < tLength) {
            if(s.charAt(l) == t.charAt(r)) {
                l++;
                r++;
            } else {
                l++;
            }
        }

        return tLength - r;
    }
}

File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        size_t l{};

        for (auto i{0uz}; i < s.size(); ++i) {
            if (l == t.size()) break;

            if (t[l] == s[i]) ++l;
        }

        return t.size() - l;
    }
};